### ISSUE TITLE
Multi-instance output element expression with space prefix leads to incorrect variable scope

### ISSUE THREAD

**Issue by korthout** [zeebe-io/zeebe#4852]
**Describe the bug**
When the output element expression of a multi-instance element directly refers to a variable or a property of a variable, a new variable with that name should be nil initialised at the multi-instance body scope. This is especially important for parallel multi-instance elements, because this is a common trap for a race condition.

Our documentation states it like (including a space in the expression):
> `outputElement` is an expression that defines the output of the instance (e.g. `= result`). Usually, it accesses a variable of the instance that holds the output value. If the expression only accesses a variable or a nested property then it is created as local variable of the instance. 

However, the local variable is wrongly initialised when the expression is prefixed with a space.

**To Reproduce**
- deploy a simple workflow with a multi-instance service task
  - input collection `= inputs`
  - input element `input`
  - output collection `outputs`
  - output element `= output`
- create an instance with variable: `inputs:[1,2]`
- create a java worker that prints the variables of an activated job and completes it with `Map.of("output", input)`, like:
```java
final Map<String, Object> variables = job.getVariablesAsMap();
final var input = variables.get("input");
System.out.println(new ObjectMapper().writeValueAsString(variables));
client.newCompleteCommand(job.getKey())
         .variables(Map.of("output", input))
         .send()
         .join();
```
You'll note that the worker print the output variable with a space prefixed to the variable name: 
`{"input":1," output":null,"loopCounter":1,"outputs":[null,null],"inputs":[1,2]}`. 

You should also note that the `output` variable exists at the process scope instead of at the task scope.

**Expected behavior**
Output element expressions that directly refer to a variable or a property of a variable, with a space in the expression should be correctly nil initialised.

**Log/Stacktrace**
<details><summary>Full Stacktrace</summary>
 <p>

```
# Run with outputElement expression `= output`
{"input":1," output":null,"loopCounter":1,"outputs":[null,null],"inputs":[1,2]}
{"input":2," output":null,"loopCounter":2,"outputs":[null,null],"inputs":[1,2]}

# Run with outputElement expression `=output`
{"input":1,"output":null,"loopCounter":1,"outputs":[null,null],"inputs":[1,2]}
{"input":2,"output":null,"loopCounter":2,"outputs":[null,null],"inputs":[1,2]}
```

</p>
</details>

**Environment:**
 - OS: macOS 10.15.5, but shouldn't matter
 - Zeebe Version: 0.23.3
 - Configuration: default + Hazelcast exporter, but shouldn't matter

**Comment by korthout** [comment]
A workaround exists by either:
- not using a space as prefix in the expression
- declaring an input mapping on the task to explicitly nil initialise: source `= null`, target `output`

**Comment by strawhat5** [comment]
Thanks for opening this @korthout !

**Comment by saig0** [comment]
This bug is already fixed since `0.24.0-alpha1` by @korthout itself with #4100 :smile:

**Comment by korthout** [comment]
> This bug is already fixed since `0.24.0-alpha1` by @korthout itself with #4100 😄

Oops. I did not check, but merely assumed that it was already fixed in `0.23.1` and that this specific situation was different. 😄   The lesson for me is to always verify a bug exists on latest build first. Thanks for this check @saig0
